src/Connections/GithubRepo.ts

```diff
@@ -115,46 +204,29 @@ function compareEmojiStrings(e0: string, e1: string, e0Index = 0) {
     return e0.codePointAt(e0Index) === e1.codePointAt(0);
 }
 
-function validateState(state: Record<string, unknown>): GitHubRepoConnectionState {
-    if (typeof state.org !== "string") {
-        throw new ApiError("Expected a 'org' property", ErrCode.BadValue);
-    }
-    if (typeof state.repo !== "string") {
-        throw new ApiError("Expected a 'repo' property", ErrCode.BadValue);
-    }
-    const res: GitHubRepoConnectionState = {
-        org: state.org,
-        repo: state.repo,
-    }
-    if (state.commandPrefix) {
-        if (typeof state.commandPrefix !== "string") {
-            throw new ApiError("Expected 'commandPrefix' to be a string", ErrCode.BadValue);
-        }
-        if (state.commandPrefix.length < 2 || state.commandPrefix.length > 24) {
-            throw new ApiError("Expected 'commandPrefix' to be between 2-24 characters", ErrCode.BadValue);
-        }
-        res.commandPrefix = state.commandPrefix;
-    }
-    if (state.ignoreHooks && Array.isArray(state.ignoreHooks)) {
-        if (state.ignoreHooks?.find((ev) => !AllowedEvents.includes(ev))?.length) {
-            throw new ApiError(`'events' can only contain ${AllowedEvents.join(", ")}`, ErrCode.BadValue);
-        }
-        res.ignoreHooks = state.ignoreHooks;
-    }
-    return res;
-}
-
-
 /**
  * Handles rooms connected to a github repo.
  */
 @Connection
 export class GitHubRepoConnection extends CommandConnection implements IConnection {
+
+	static validateState(state: Record<string, unknown>, isExistingState = false): GitHubRepoConnectionState {
+        const validator = new Ajv().compile(ConnectionStateSchema);
+        if (validator(state)) {
+            // Validate ignoreHooks IF this is an incoming update (we can be less strict for existing state)
```

Comment:
Are we only less strict then because of backwards compatibility?

Reply:
This isn't covered by the new tests (regardless of `isExistingState`) – would be nice to have it there, and it'd be a good place for the rationale for the future reader :)

[instanbul](https://istanbul.js.org/) confirms:

![image](https://user-images.githubusercontent.com/86378/167775496-09f6ad1f-1f3f-4448-87b6-2e0292076f9c.png)

Reply:
>Are we only less strict then because of backwards compatibility?

Yeah, though I'm wondering if it's even worth being strict here.